app/src/main/java/com/mapbox/search/demo/SearchViewBottomSheetsMediator.kt

```diff
@@ -131,23 +131,27 @@ class SearchViewBottomSheetsMediator(
             if (transaction == null) {
                 fallback { "Transaction is null" }
             } else {
-                when (transaction.screen) {
-                    Screen.CATEGORIES -> {
-                        val category = (transaction.arg as? Bundle)?.unwrapCategory()
-                        if (category == null) {
-                            fallback { "Saved category is null" }
-                        } else {
-                            openCategory(category, fromBackStack = true)
-                        }
-                    }
-                    Screen.PLACE -> {
-                        val place = transaction.arg as? SearchPlace
-                        if (place == null) {
-                            fallback { "Saved place is null" }
-                        } else {
-                            openPlaceCard(place, fromBackStack = true)
-                        }
-                    }
+                transaction.execute()
+            }
+        }
+    }
+
+    private fun Transaction.execute() {
```

Comment:
nit
As for me, it doesn't seem like a good use case for the Kotlin-extension function, even though it's private in a class. It looks misleading in this case. What this function does is very specific logic related to the screen type, on the other hand, `Transaction.execute` looks like something generic, probably which applies pending job or something like that.

Reply:
Discussed internally. Agreed to fix this issue in next release.